From 155d4b167bfef89f81fc0e5cb7a4a0940ba2b4a8 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Tue, 9 Apr 2019 10:00:42 +0800
Subject: [PATCH 15/30] media: imx8: use new structure to avoid fwnode.fwnode

In upstream commit 4e48afecd5ee ("media: v4l2-async: simplify
v4l2_async_subdev structure"), fwnode.fwnode is not needed anymore.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/media/platform/imx8/mxc-media-dev.c       | 6 +++---
 drivers/media/platform/imx8/mxc-mipi-csi2_yav.c   | 4 ++--
 drivers/media/platform/mxc/capture/mxc_mipi_csi.c | 4 ++--
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-media-dev.c b/drivers/media/platform/imx8/mxc-media-dev.c
index 344748c2b52c..6fca4ff7eb78 100644
--- a/drivers/media/platform/imx8/mxc-media-dev.c
+++ b/drivers/media/platform/imx8/mxc-media-dev.c
@@ -286,7 +286,7 @@ static int subdev_notifier_bound(struct v4l2_async_notifier *notifier,
 	dev_dbg(&mxc_md->pdev->dev, "%s\n", __func__);
 	/* Find platform data for this sensor subdev */
 	for (i = 0; i < ARRAY_SIZE(mxc_md->sensor); i++) {
-		if (mxc_md->sensor[i].asd.match.fwnode.fwnode ==
+		if (mxc_md->sensor[i].asd.match.fwnode ==
 				of_fwnode_handle(sd->dev->of_node))
 			sensor = &mxc_md->sensor[i];
 	}
@@ -355,7 +355,7 @@ static int register_sensor_entities(struct mxc_md *mxc_md)
 
 		if (!of_node_cmp(node->name, "hdmi_rx")) {
 			mxc_md->sensor[index].asd.match_type = V4L2_ASYNC_MATCH_FWNODE;
-			mxc_md->sensor[index].asd.match.fwnode.fwnode = of_fwnode_handle(node);
+			mxc_md->sensor[index].asd.match.fwnode = of_fwnode_handle(node);
 			mxc_md->async_subdevs[index] = &mxc_md->sensor[index].asd;
 
 			mxc_md->num_sensors++;
@@ -401,7 +401,7 @@ static int register_sensor_entities(struct mxc_md *mxc_md)
 		}
 
 		mxc_md->sensor[index].asd.match_type = V4L2_ASYNC_MATCH_FWNODE;
-		mxc_md->sensor[index].asd.match.fwnode.fwnode = of_fwnode_handle(rem);
+		mxc_md->sensor[index].asd.match.fwnode = of_fwnode_handle(rem);
 		mxc_md->async_subdevs[index] = &mxc_md->sensor[index].asd;
 
 		mxc_md->num_sensors++;
diff --git a/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c b/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
index df5710aa186b..0c97ff990c73 100644
--- a/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
+++ b/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
@@ -501,7 +501,7 @@ static int subdev_notifier_bound(struct v4l2_async_notifier *notifier,
 	struct mxc_mipi_csi2_dev *csi2dev = notifier_to_mipi_dev(notifier);
 
 	/* Find platform data for this sensor subdev */
-	if (csi2dev->asd.match.fwnode.fwnode == of_fwnode_handle(subdev->dev->of_node))
+	if (csi2dev->asd.match.fwnode == of_fwnode_handle(subdev->dev->of_node))
 		csi2dev->sensor_sd = subdev;
 
 	if (subdev == NULL)
@@ -546,7 +546,7 @@ static int mipi_csis_subdev_host(struct mxc_mipi_csi2_dev *csi2dev)
 				  port->full_name);
 
 		csi2dev->asd.match_type = V4L2_ASYNC_MATCH_FWNODE;
-		csi2dev->asd.match.fwnode.fwnode = of_fwnode_handle(rem);
+		csi2dev->asd.match.fwnode = of_fwnode_handle(rem);
 		csi2dev->async_subdevs[0] = &csi2dev->asd;
 
 		of_node_put(rem);
diff --git a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
index 30860ac6610b..e491735b20e9 100644
--- a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
+++ b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
@@ -943,7 +943,7 @@ static int subdev_notifier_bound(struct v4l2_async_notifier *notifier,
 	struct csi_state *state = notifier_to_mipi_dev(notifier);
 
 	/* Find platform data for this sensor subdev */
-	if (state->asd.match.fwnode.fwnode == dev_fwnode(subdev->dev))
+	if (state->asd.match.fwnode == dev_fwnode(subdev->dev))
 		state->sensor_sd = subdev;
 
 	if (subdev == NULL)
@@ -1023,7 +1023,7 @@ static int mipi_csis_subdev_host(struct csi_state *state)
 		}
 
 		state->asd.match_type = V4L2_ASYNC_MATCH_FWNODE;
-		state->asd.match.fwnode.fwnode = of_fwnode_handle(rem);
+		state->asd.match.fwnode = of_fwnode_handle(rem);
 		state->async_subdevs[0] = &state->asd;
 
 		of_node_put(rem);
-- 
2.17.1

